packages/manager/apps/dedicated/client/app/components/dedicated-cloud/datacenter/add/dedicatedCloud-datacenter-add.controller.js

```diff
@@ -38,33 +42,86 @@ export default class {
           )}: ${err.message || ''}`,
           'danger',
         );
+      });
+  }
+
+  calculateEstimate() {
+    this.DedicatedCloud.getCartServiceOption(this.serviceName)
+      .then((cartServiceOption) => {
+        this.commercialRange.list = this.commercialRange.list.map(
+          (datacenter) => {
+            const totalPrices =
+              datacenter.addons?.reduce(
+                (accumulator, { planCode, quantity }, index) =>
+                  accumulator +
+                  (cartServiceOption
+                    .find((option) => option.planCode === planCode)
+                    ?.prices.reduce(
+                      (childAccumulator, { price, duration, pricingMode }) => {
+                        if (planCode.includes('-vdc')) {
```

Comment:
What's the meaning for "vdc" ?

Reply:
vdc is virtual data center, it's for find the primary planCode for the express order funnel URL.

Reply:
ok thx